contracts/src/CrossChainApplications/VerifiedBlockHash/BlockHashPublisher.sol

```diff
@@ -7,16 +7,17 @@ pragma solidity 0.8.18;
 
 import "../../Teleporter/ITeleporterMessenger.sol";
 import "../../Teleporter/upgrades/TeleporterRegistry.sol";
-import "../../Teleporter/upgrades/TeleporterUpgradeable.sol";
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "./BlockHashReceiver.sol";
 
 /**
  * Contract that publishes the latest block hash of current chain to another chain.
  */
-contract BlockHashPublisher is TeleporterUpgradeable {
+contract BlockHashPublisher {
```

Comment:
Why did you remove `TeleporterUpgradeable` from this contract?

Reply:
The `TeleporterUpgradeable` contract is mostly for the modifier `onlyAllowedTeleporter` and this contract doesn't receive Teleporter messages, only sends them, so don't think it's needed. Now that I look back at it again though, might not hurt to add back in for consistency.

Reply:
Oh yea and also inheriting `TeleporterUpgradeable` also means the contract needs to override `updateMinTeleporterVersion` and I think that's unnecessary for contracts that only send.

Reply:
Ah gotcha, that makes sense. I'm fine leaving `TeleporterUpgradeable` out.